### ISSUE TITLE
influxdb reporter is not pushing data

### ISSUE THREAD

**Issue by bpatelcs** [omergertel/pyformance#30]
Hi, 

I just wrote simple script to get me started on pyformance, seems like influxdb reporter is not pushing any data to influxdb, my script is as below, can someone let me know if something is wrong with it?

```
from pyformance.registry import MetricsRegistry
from pyformance.reporters.influx import InfluxReporter
import time

registry = MetricsRegistry()
counter=registry.counter("test")

reporter=InfluxReporter(registry=registry)
reporter.start()

while True:
    time.sleep(2)
    counter.inc()
    print counter.get_count()


```

**Comment by jerluc** [comment]
@bpatelcs can you give any more information/context? For example:

* Are you running a local or remote InfluxDB instance?
* Have you created the default `metrics` database in your InfluxDB instance?
* What is the behavior of the above script? Does it encounter a `RuntimeError` or just no output?

**Comment by vtsatskin** [comment]
I tried running the original code @bpatelcs showed using python 3.5.2, and I got the same issue: InfluxDB was not being written to by the reporter. Using python 2.7.10, the reporting successfully works. 

I believe the issue is due to the [removal of the `iteritems` method on dictionaries in python 3](http://stackoverflow.com/a/13998534). Looking into how other reporters have handled this, they use the six library for ensuring compatibility between python 2 and 3. Doing something like what [SysLogReporter](https://github.com/omergertel/pyformance/blob/32f5f1d0ca83bb7ef4f3660b1cdf85a7294d1c92/pyformance/reporters/syslog_reporter.py#L54) does for metrics iteration should fix the problem.

## Testing
```python
# metrics.py
import time
from pyformance import MetricsRegistry
from pyformance.reporters.influx import InfluxReporter

registry = MetricsRegistry()
reporter = InfluxReporter(registry, reporting_interval=1)
reporter.start()

counter = registry.counter("test.counter")

while True:
    time.sleep(2)
    counter.inc()
    print(counter.get_count())
```

```
➜  python_helllo_world git:(master) ✗ python3 ./metrics.py
1
2
3
4
5
6
7
8
9
10
^CTraceback (most recent call last):
  File "./metrics.py", line 12, in <module>
    time.sleep(2)
KeyboardInterrupt
```

Trying to force reporting yields an error.
```python
# metrics2.py
from pyformance import MetricsRegistry
from pyformance.reporters.influx import InfluxReporter

registry = MetricsRegistry()
reporter = InfluxReporter(registry, reporting_interval=1)
reporter.start()

counter = registry.counter("test.counter")
counter.inc()
print("Counter value: ", counter.get_count())
print("Metrics registry: ", registry.dump_metrics())
print("Forcing reporting...")
reporter.report_now()
```

```
➜  python_helllo_world git:(master) ✗ python3 ./metrics2.py
Counter value:  1
Metrics registry:  {'test.counter': {'count': 1}}
Forcing reporting...
Traceback (most recent call last):
  File "./metrics.py", line 13, in <module>
    reporter.report_now()
  File "/usr/local/lib/python3.5/site-packages/pyformance/reporters/influx.py", line 72, in report_now
    for key, metric_values in metrics.iteritems():
AttributeError: 'dict' object has no attribute 'iteritems'
```

Running the above code using python 2.7.10 yields a successful write:
```
➜  python_helllo_world git:(master) ✗ python2 ./metrics.py
('Counter value: ', 1)
('Metrics registry: ', {'test.counter': {'count': 1}})
Forcing reporting...
```

```
> use metrics
Using database metrics
> show measurements
name: measurements
name
----
test.counter
```

## InfluxDB Setup
```
~ influx
Connected to http://localhost:8086 version v1.2.3
InfluxDB shell version: v1.2.3
> show databases
name: databases
name
----
_internal
metrics
```

Snippet from `influxd config`
```
[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = false
  log-enabled = true
  write-tracing = false
  pprof-enabled = true
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"
  https-private-key = ""
  max-row-limit = 0
  max-connection-limit = 0
  shared-secret = ""
  realm = "InfluxDB"
  unix-socket-enabled = false
  bind-socket = "/var/run/influxdb.sock"
```

**Comment by vtsatskin** [comment]
Correct me if I'm wrong here, but I believe this would have been caught if a test existed for checking the `report_now` method. Looks like the test for the reporter just checks to see if the initializer doesn't error out.

**Comment by omergertel** [comment]
I'm not using influxdb myself. If you can make a pull request that fixes this issue (and the test also), I'd be happy to merge it.

**Comment by vtsatskin** [comment]
@omergertel I made the fixes required, however, I ended up getting some issues with with using string versus byte strings and urllib (sorry I can't remember the exact issue). So to resolve this, either I have to figure out the problems with urllib, or use the [influxdb wrapper](https://github.com/influxdata/influxdb-python). I would prefer to use the wrapper as I'm not an expert on the protocol and it seems error prone to build the HTTP requests manually. Would it be acceptable to introduce this new dependency into the project?
